<launch>
    <arg name="robot_name" default="strirus_gamma"/>
    <arg name="robot_base_frame" default="base_footprint"/>
    <arg name="world_frame" default="world"/>

    <param name="robot_description"
           command="$(find xacro)/xacro '$(find robot_description)/xarco/strirus_gamma.xacro'"/>

    <include file="$(find robot_description)/launch/strirus_robot.xml">
        <arg name="sim" value="1"/>
    </include>

    <arg name="paused" default="true"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="true"/>
    <!-- <arg name="world_name" default="$(find world_description)/worlds/testing_area.world"/> -->
  
    <node name="start_coppeliasim" pkg="coppeliasim_run" type="start_coppeliasim" args=" -s $(find sim_perception_map_first_iter)/scene/strirus.ttt"  output="screen"/>


    <node name="ideal_robot_localization" pkg="sim_perception_map_first_iter" type="ideal_robot_localization.py"
          output="screen" required="false">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="robot_base_frame" value="$(arg robot_base_frame)"/>
        <param name="world_frame" value="$(arg world_frame)"/>
        <param name="debug" value="$(arg debug)"/>
    </node>

    <arg name="rvizconfig" default="$(find sim_perception_map_first_iter)/rviz/rviz_config.rviz" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />


</launch>